kratos/utilities/compare_elements_and_conditions_utility.h

```diff
@@ -39,39 +38,19 @@ class CompareElementsAndConditionsUtility
     ///@name Operations
     ///@{
 
-    static void GetRegisteredName(const Element& rElement, std::string& rName) {
-        KRATOS_TRY;
+    static void KRATOS_API(KRATOS_CORE) GetRegisteredName(const Geometry<Node<3>>& rGeometry, std::string& rName);
 
-        for(auto const& component: KratosComponents<Element>::GetComponents()) {
-            if (GeometricalObject::IsSame(*(component.second), rElement)) {
-                rName = component.first;
-                return;
-            }
-        }
-
-        KRATOS_ERROR << "Element \"" << typeid(rElement).name() << "\" not found!" << std::endl;
-
-        KRATOS_CATCH("");
+    static void GetRegisteredName(const Geometry<Node<3>>* pGeometry, std::string& rName) {
+        CompareElementsAndConditionsUtility::GetRegisteredName(*pGeometry, rName);
     }
 
+    static void KRATOS_API(KRATOS_CORE) GetRegisteredName(const Element& rElement, std::string& rName);
+
     static void GetRegisteredName(const Element* pElement, std::string& rName) {
         CompareElementsAndConditionsUtility::GetRegisteredName(*pElement, rName);
     }
 
-    static void GetRegisteredName(const Condition& rCondition, std::string& rName) {
-        KRATOS_TRY;
-
-        for(auto const& component: KratosComponents<Condition>::GetComponents()) {
-            if (GeometricalObject::IsSame(*(component.second), rCondition)) {
-                rName = component.first;
-                return;
-            }
-        }
-
-        KRATOS_ERROR << "Condition \"" << typeid(rCondition).name() << "\" not found!" << std::endl;
-
-        KRATOS_CATCH("");
-    }
+    static void KRATOS_API(KRATOS_CORE) GetRegisteredName(const Condition& rCondition, std::string& rName);
```

Comment:
```suggestion
    static void KRATOS_API(KRATOS_CORE) GetRegisteredName(
        const Condition& rCondition,
        std::string& rName);
```

(minor style guide suggestion)